src/main/resources/avro/alleleAnnotations.avdl

```diff
@@ -0,0 +1,190 @@
+Variant.
+
+The TranscriptEffect sub record holds information on the effect of a specific
+allele on a specific transcript.
+As Variants may overlap multiple transcripts, they may have multiple
+TranscriptEffect records. Variants with multiple alternate alleles will have
+multiple TranscriptEffect records per transcript.
+(2 alternate alleles x 3 transcripts = 6 TranscriptEffect records)
+
+VariantAnnotation records belong to VariantAnnotationSets.
+VariantAnnotationSets are created by comparing a number of Variants from a
+VariantSet to a specific set of reference data using specific software tools.
+A VariantAnnotationSet contains information on reference data and software
+versions used in calculating the annotation; it is essential this information
+is exhaustive.
+
+*/
+
+/**
+An AnalysisResult record holds the output of a prediction package such
+as SIFT on a specific allele.
+*/
+record AnalysisResult {
+
+  /** The analysis record for this result (defined in metadata schema)  */
+  Analysis analysis;
+
+  /** The text-based result for this analysis */
+  union { null, string} analysisResult;
+
+  /** The numeric score for this analysis  */
+  union { null, int} analysisScore;
+
+}
+
+/**
+An allele location record holds the location of an allele relative to a
+non-genomic coordinate system such as a CDS or protein and holds the
+reference and alternate sequence where appropriate
+*/
+record AlleleLocation {
+
+  /** Relative start position of the allele in this coordinate system */
+  int overlapStart ;
+
+  /** Relative end position of the allele in this coordinate system  */
+  union { null, int } overlapEnd = null;
+
+  /** Reference sequence in feature (this should be the codon at CDS level) */
+  union { null, string } referenceSequence = null;
+
+  /** Alternate sequence in feature (this should be the codon at CDS level) */
+  union { null, string } alternateSequence = null;
+}
+
+/**
+Impact is a simple prioritization for the effect of an allele which is used
+in the annotation record.
+IMPORTANT:
+ Prioritization methods are a crude estimates and are not assumed to be
+ reliable: a 'HIGH' Impact may actually not cause any disruption
+ in protein function or expression.
+*/
+enum Impact {
+ HIGH,      // Variant highly disrupts protein function
+ MODERATE,  // Moderately disrupts protein function
+ LOW,       // Low disruption of protein impact
+ MODIFIER   // No known effect
+}
+
+/**
+A VariantAnnotationSet record groups VariantAnnotation records. It is derived
+from a VariantSet and holds information describing the software and reference
+data used in the annotation.
+*/
+record VariantAnnotationSet {
+
+  /** The ID of the variant annotation set record */
+  string id;
+
+  /** The ID of the variant set to which this annotation set belongs */
+  string variantSetId;
+
+  /**
+  Analysis details. It is essential to supply versions for all software and
+  reference data used.
+  */
+  Analysis analysis;
```

Comment:
OK, fair enough. I guess we should provide a GET endpoint for the Analysis objects as well though, since we clients will have access to the IDs? It seems unnecessary to provide a full search endpoint for them.